core/src/main/java/org/jdbi/v3/GeneratedKeys.java

```diff
@@ -74,4 +77,9 @@
             throw new ResultSetException("Exception thrown while attempting to traverse the result set", e, context);
         }
     }
+
+    @Override
+    public <ContainerType> ContainerType collect(Class<ContainerType> containerType) {
+        return stream().collect(containerFactoryRegistry.createCollectorFor(containerType));
+    }
```

Comment:
Makes sense. 

I also thought about using `java.lang.reflect.Type` as the parameter type and resolving the actual type on runtime. With that users can write type-safe code like 
```java
ImmutableList<String> rs = h.createQuery("select name from something order by id")
                .mapTo(String.class)
                .collect(new TypeToken<ImmutableList<String>>(){}.getType());
```

What do you think? Is it worth it?